internal/argocd/revision_test.go

```diff
@@ -5,30 +5,40 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
 
 	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	argocd "github.com/akuity/kargo/internal/controller/argocd/api/v1alpha1"
 	argocdapi "github.com/akuity/kargo/internal/controller/argocd/api/v1alpha1"
 )
 
-func TestGetDesiredRevision(t *testing.T) {
+func TestGetDesiredRevisions(t *testing.T) {
 	testOrigin := kargoapi.FreightOrigin{
 		Kind: kargoapi.FreightOriginKindWarehouse,
 		Name: "fake-warehouse",
 	}
+	testOrigin2 := kargoapi.FreightOrigin{
+		Kind: kargoapi.FreightOriginKindWarehouse,
+		Name: "another-warehouse",
+	}
+
+	var no_revisions []string
```

Comment:
Apart from the deviation from standard Go variable naming conventions here, I'm not sure I would declare a variable for this at all. The value of this variable is forever `nil` and all the individual test cases read more cleanly if they just _say_ they expect `nil`.

Nit: That being said, expecting `nil` may even unnecessarily brittle. It works now, but very well may break the moment someone refactors the function under test. The important thing we want to assert in many of these cases is that what is returned is _empty_, which could be `nil` or it could be an non-nil, but still empty slice. In the majority of cases, I find it's useful to have an assertion function as part of each test to perform case-specific assertions. For these cases, you'd assert something like `require.Empty(t, result)`.